File: js/cgol.js
const minimist = require('minimist');
const World = require("./world.js");

const args = minimist(process.argv.slice(2));

const width = args.width || 30;
const height = args.height || 30;
const liveCount = args['live-count'] || 100;
const stepTime = args['step-time'] || 250;
// Not supporting seed like other implementations because JS doesn't natively suport it.

function render(world) {
  process.stdout.write("\x1Bc");

  for (let y = 0; y < world.height; y++) {
    for (let x = 0; x < world.width; x++) {
      process.stdout.write(world.get(x, y) ? "*" : " ");
    }
    process.stdout.write("\n");
  }
}

let w = new World(width, height);

for (let i = 0; i < liveCount; i++) {
  w.set(Math.floor(Math.random() * width), Math.floor(Math.random() * height), true);
}

function mainLoop() {
  render(w);
  w = w.step();
  setTimeout(mainLoop, stepTime);
}

mainLoop();

File: ruby/cgol.rb
require 'optparse'
require_relative "world"

options = {}
OptionParser.new do |parser|
  parser.banner = "Usage: cgol.rb [options]"

  parser.on("--width INTEGER", Integer, "Width") do |value|
    options[:width] = value
  end
  parser.on("--height INTEGER", Integer, "Height") do |value|
    options[:height] = value
  end
  parser.on("--live-count INTEGER", Integer, "Live cell count to start") do |value|
    options[:live_count] = value
  end
  parser.on("--step-time INTEGER", Integer, "Step time in milliseconds") do |value|
    options[:step_time] = value
  end
  parser.on("--seed INTEGER", Integer, "Seed for RNG") do |value|
    options[:seed] = value
  end
end.parse!

width = options.fetch(:width, 30)
height = options.fetch(:height, 30)
live_count = options.fetch(:live_count, 100)
step_time = options.fetch(:step_time, 250).to_f / 1000
seed = options.fetch(:seed, 0)
srand(seed)

def render(world)
  puts "\033c"

  world.height.times do |y|
    world.width.times do |x|
      print world.get(x, y) ? "*" : " "
    end
    puts
  end
end

w = World.new(width, height)

live_count.times do
  w.set(rand(w.width), rand(w.height), true)
end

loop do
  render(w)
  w = w.step
  sleep(step_time)
end
